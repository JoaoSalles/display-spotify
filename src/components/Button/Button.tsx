import React from "react";
import { Button as BaseButton, SHAPE } from "baseui/button";

interface Props {
  onClick: () => void
}

function Button(props: Props) {
  const { onClick } = props;

  return (
    <BaseButton
      onClick={() => onClick()}
      shape={SHAPE.pill}
    >
      Hello
    </BaseButton>
  );
}

export default Button;