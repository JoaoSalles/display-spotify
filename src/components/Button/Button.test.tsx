import React from "react";
import { render, fireEvent, waitFor } from "@testing-library/react";
import Button from "./Button";

describe("Tests for Button", () => {
    it("Should match snapshot", async () => {
        const { container } = render(<Button onClick={() => {}} />);

        expect(container).toMatchSnapshot();
    });

    it("Should render button and trigger it", async () => {
        const mockClick = jest.fn();
        const { getByTestId } = render(<Button onClick={mockClick} />);

        const buttonField = await waitFor(() => getByTestId("button-test"));

        fireEvent.click(buttonField);

        expect(mockClick).toBeCalled();
    });
});
