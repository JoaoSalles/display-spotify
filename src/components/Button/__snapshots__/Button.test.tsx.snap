// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests for Button Should match snapshot 1`] = `
<div>
  <button
    class=""
    data-baseweb="button"
    data-testid="button-test"
  >
    Logar com Spotify
  </button>
</div>
`;
